### PR TITLE
qt/5.15.3: fix qtwebengine with latest glibc

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qt/5.15.3**

fixes conan-io/conan-center-index#10068

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — DISMISSED** [review]

**Review by uilianries — DISMISSED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by MartinDelille — DISMISSED** [review]

**Review by uilianries — APPROVED** [review]

File: recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch [hunk 1]
```diff
@@ -0,0 +1,10 @@
+diff --git a/chromium/third_party/skia/src/utils/SkParseColor.cpp b/chromium/third_party/skia/src/utils/SkParseColor.cpp
+index 7260365b2c6..96f46dd67f9 100644
+--- a/chromium/third_party/skia/src/utils/SkParseColor.cpp
++++ b/chromium/third_party/skia/src/utils/SkParseColor.cpp
+@@ -9,3 +0,5 @@
+ #include "include/utils/SkParse.h"
+ 
++#include <algorithm> // std::lower_bound
```
Comment (jwillikers) [inline recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch:8]: You also need an include here for `<iterator>` in order to include `std::begin` and `std::end`. Without that, I hit a compilation error with GCC 12.

**Review by jwillikers — CHANGES_REQUESTED** [review]

File: recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch (hunk 1 above)
Comment (ericLemanissier) [reply recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch:8]: I'm sorry, but I find this too bleeding edge. gcc 12 is not even released. AFAIK, this fix is not submitted to skia.
Each patch we add to a recipe is a burden we have to carry during the next releases, with the risk that it conflicts with other changes upstream. This is why using patches of commits already merged upstream is the least bad solution: there is a minimal chance of conflict, and we can very easily check in which version the patch is integrated, to drop it as soon as necessary.
also, if you absolutely need to be bleeding edge, then don't use a version of qt which is more than 1 year old ([5.15.3](https://github.com/qt/qtbase/commits/v5.15.3-lts-lgpl)). you should instead use 6.3.0.

Comment (jwillikers) [reply recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch:8]: I submitted a patch upstream which has now been merged: https://skia-review.googlesource.com/c/skia/+/533981

Comment (ericLemanissier) [reply recipes/qt/5.x.x/patches/chromium-skia-missing-iterator-include.patch:8]: thanks, this will ease maintenance !

**Review by jwillikers — APPROVED** [review]

**Review by MartinDelille — APPROVED** [review]

**Review by uilianries — APPROVED** [review]

**Pull request merged**